### PR TITLE
Fix duplicate logs and errors codes

### PR DESCRIPTION
**Pull request by Avi-Eyal**
### What does this PR do?
- _This PR includes minor fix to certain duplicate log codes and error codes in conjur_

#### Change log
- [ ] The CHANGELOG has been updated, or
- [X] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [X] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [X] This PR does not require updating any documentation

#### API Changes
- [ ] The [OpenAPI spec](https://github.com/cyberark/conjur-openapi-spec) has been updated to meet new API changes (or an issue has been opened), or
- [X] The changes in this PR do not affect the Conjur API

### CODE REVIEW

**Review by tzheleznyak — APPROVED** [review]

File: app/domain/errors.rb [hunk 1]
```diff
@@ -360,7 +360,7 @@ module AuthnGcp
       InvalidAccountInAudienceClaim = ::Util::TrackableErrorClass.new(
         msg: "'audience' token claim '{0}' is invalid. " \
               "The account in the audience '{1}' does not match the account in the URL request '{2}'",
-        code: "CONJ00071E"
+        code: "CONJ00070E"
```
Comment (tzheleznyak) [inline app/domain/errors.rb:363]: @hilagross
How can we insure the deleted code not appears somewhere in the documentation?

File: app/domain/logs.rb [hunk 2]
```diff
@@ -121,7 +121,7 @@ module ResourceRestrictions
 
       ExtractingRestrictionsFromResource = ::Util::TrackableLogMessageClass.new(
         msg: "Extracting resource restrictions for authenticator '{0-authn-name}' from host '{1-host-name}'",
-        code: "CONJ00039D"
+        code: "CONJ00032D"
       )
```
Comment (tzheleznyak) [inline app/domain/logs.rb:126]: @hilagross
How can we insure the deleted code not appears somewhere in the documentation?

Comment (hilagross) [reply app/domain/logs.rb:126]: @tzheleznyak We checked docsumentation it is not in there.
@shulifink please make sure it wasn't added lately to our docs.

**Review by nessiLahav — APPROVED** [review]
LGTM

File: cucumber/authenticators_gcp/features/authn_gce.feature [hunk 3]
```diff
@@ -104,7 +104,7 @@ Feature: GCP Authenticator - GCE flow, hosts can authenticate with GCP authentic
     Then it is unauthorized
     And The following matches the log after my savepoint:
     """
-    CONJ00071E 'audience' token claim .* is invalid. The account in the audience .* does not match the account in the URL request .*
+    CONJ00070E 'audience' token claim .* is invalid. The account in the audience .* does not match the account in the URL request .*
```
Comment (shulifink) [inline cucumber/authenticators_gcp/features/authn_gce.feature:107]: @hilagross  - am I supposed to check this?
What is .*??

File: app/domain/errors.rb (hunk 1 above)
Comment (shulifink) [reply app/domain/errors.rb:363]: @tzheleznyak  - The account in audience '{1}' does not match the account in URL request '{2}'"

File: app/domain/logs.rb [hunk 4]
```diff
@@ -121,7 +121,7 @@ module ResourceRestrictions
 
       ExtractingRestrictionsFromResource = ::Util::TrackableLogMessageClass.new(
         msg: "Extracting resource restrictions for authenticator '{0-authn-name}' from host '{1-host-name}'",
-        code: "CONJ00039D"
+        code: "CONJ00032D"
```
Comment (shulifink) [inline app/domain/logs.rb:124]: Is this one of our (CyberArk) authenticators?
Can you give me an example of filled in....


If one of ours, then,  

Extracting resource restrictions for {0-authn-name} Authenticator from host '{1-host-name}'",
for example: 
"Extracting resource restrictions for Azure Authenticator from host '{1-host-name}'",

**Pull request merged**